notebooks/Chapter_5-Parameter_Efficient_Fine_tuning_Tutorial.ipynb

```diff
@@ -0,0 +1,8464 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "colab_type": "text",
+    "id": "view-in-github"
+   },
+   "source": [
+    "<a href=\"https://colab.research.google.com/github/garrettsomers/ManningLLMBook/blob/lora_tutorial/LoRA_Tutorial.ipynb\" target=\"_parent\"><img src=\"https://colab.research.google.com/assets/colab-badge.svg\" alt=\"Open In Colab\"/></a>"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "colab": {
+     "base_uri": "https://localhost:8080/"
+    },
+    "id": "ng90osNYKJHg",
+    "outputId": "2fecaaee-d0ff-4b76-c5b4-d6b7f74c07f7"
+   },
+   "outputs": [],
+   "source": [
+    "! huggingface-cli login"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "colab": {
+     "base_uri": "https://localhost:8080/"
+    },
+    "id": "n7MhgnrrbWTL",
+    "outputId": "1d25623f-9b18-4e39-a92a-2149bc0e0224"
+   },
+   "outputs": [],
+   "source": [
+    "#!pip install -U sentencepiece\n",
+    "!pip install -q kaggle torch peft==0.4.0 bitsandbytes==0.40.2 transformers==4.31.0 trl==0.4.7 accelerate openai\n",
+    "!git clone https://github.com/guyfe/Tweetsumm.git"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "id": "vLDzVeGp5Zig"
+   },
+   "outputs": [],
+   "source": [
+    "## tweet_sum_processor.py has an issue where it crashes when < 3 summaries are\n",
+    "## available. add a small fix:\n",
+    "with open('Tweetsumm/tweet_sum_processor.py', 'r') as file:\n",
+    "    lines = file.readlines()\n",
+    "lines.insert(79, \"            if summ is None: continue\\n\")\n",
+    "with open('Tweetsumm/tweet_sum_processor2.py', 'w') as file:\n",
+    "    file.writelines(lines)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "id": "oVxgCCPu7YB6"
+   },
+   "outputs": [],
+   "source": [
+    "from datasets import Dataset, DatasetDict\n",
+    "import numpy as np\n",
+    "import pandas as pd\n",
+    "import json\n",
+    "import os\n",
+    "import matplotlib.pyplot as plt\n",
+    "import ast\n",
+    "\n",
+    "%matplotlib inline"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "id": "5iAbsZjqqG_i"
+   },
+   "source": [
+    "# Collect and format data"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "colab": {
+     "base_uri": "https://localhost:8080/"
+    },
+    "id": "3_9FUeVxrhkp",
+    "outputId": "67d55c00-b5aa-4dd4-c5e2-60710e47721a"
+   },
+   "outputs": [],
+   "source": [
+    "## Upload your kaggle.json cred file into your home directory on Google colab.\n",
+    "os.path.exists('~/.kaggle/'):"
```

Comment:
This cell is broken and seems unneeded… did you intend to remove it?

Reply:
Good catch, I deleted that cell and tidied up the few cells at the top. In particular, I added the command "git -C Tweetsumm/ checkout 4903b0f20665a59e4b5494abd83d8735893c0333" to ensure that the downloaded Tweetsumm goes to the correct commit (in case they ever update their repo)